cpp/oneapi/dal/algo/linear_kernel/test/batch.cpp

```diff
@@ -0,0 +1,163 @@
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/linear_kernel/compute.hpp"
+
+#include "oneapi/dal/test/engine/common.hpp"
+#include "oneapi/dal/test/engine/fixtures.hpp"
+#include "oneapi/dal/test/engine/dataframe.hpp"
+#include "oneapi/dal/test/engine/math.hpp"
+
+namespace oneapi::dal::linear_kernel::test {
+
+namespace te = dal::test::engine;
+namespace la = te::linalg;
+
+template <typename TestType>
+class linear_kernel_batch_test : public te::algo_fixture {
+public:
+    using Float = std::tuple_element_t<0, TestType>;
+    using Method = std::tuple_element_t<1, TestType>;
+
+    auto get_descriptor(double scale, double shift) const {
+        return linear_kernel::descriptor<float, Method>{}.set_scale(scale).set_shift(shift);
+    }
+
+    te::table_id get_homogen_table_id() const {
+        return te::table_id::homogen<Float>();
+    }
+
+    void general_checks(const te::dataframe& x_data,
+                        const te::dataframe& y_data,
+                        double scale,
+                        double shift,
+                        const te::table_id& x_data_table_id,
+                        const te::table_id& y_data_table_id) {
+        CAPTURE(scale);
+        CAPTURE(shift);
+        const table x = x_data.get_table(x_data_table_id);
+        const table y = y_data.get_table(y_data_table_id);
+
+        INFO("create descriptor")
+        const auto linear_kernel_desc = get_descriptor(scale, shift);
+
+        INFO("run compute");
+        const auto compute_result = this->compute(linear_kernel_desc, x, y);
+        check_compute_result(scale, shift, x, y, compute_result);
+    }
+
+    void check_compute_result(double scale,
+                              double shift,
+                              const table& x_data,
+                              const table& y_data,
+                              const linear_kernel::compute_result<>& result) {
+        const auto result_values = result.get_values();
+
+        SECTION("result values table shape is expected") {
+            REQUIRE(result_values.get_row_count() == x_data.get_row_count());
+            REQUIRE(result_values.get_column_count() == y_data.get_row_count());
+        }
+
+        SECTION("there is no NaN in result values table") {
+            REQUIRE(te::has_no_nans(result_values));
+        }
+
+        SECTION("result values are expected") {
+            check_result_values(scale, shift, x_data, y_data, result_values);
+        }
+    }
+
+    void check_result_values(double scale,
+                             double shift,
+                             const table& x_data,
+                             const table& y_data,
+                             const table& result_values) {
+        auto reference = compute_reference(scale, shift, x_data, y_data);
+        const double tol = te::get_tolerance<Float>(1e-2, 1e-2);
```

Comment:
How this tolerance is determined?

Reply:
Aligned tolerance with the another tests. In kernels tests is better to use the Normal generator. Perhaps we should wait for merge to the master.